### PR TITLE
Github integration

### LINKED ISSUE #160: ✨ Open a GitHub repo as a project
**Is your feature request related to a problem? Please describe.**
_A clear and concise description of what the problem is. Ex. I'm always frustrated when ..._

Imagine being able to open a GitHub repo as the project inside the IDE, this could be used by library authors to provide an example playground for their library and in general to test things out in large codebases without having to rewrite everything from scratch

**Describe the solution you'd like**
_A clear and concise description of what you want to happen._

We could either have a query parameters or do something like stackblitz where if you go to a specific route that mimic the one on GitHub you open that project 

**Describe alternatives you've considered**
_A clear and concise description of any alternative solutions or features you've considered._

**Additional context**
_Add any other context or screenshots about the feature request here._

### PR DESCRIPTION
**Pull request by paoloricciuti**
Closes #160 

This still need some work @SarcevicAntonio, i want a prettier splash screen while we wait for the github repo to load and probably a better retry strategy + error handling in case we get rate-limited

Key features:

- Adds a way to load a github repo into sveltelab by specifying search params owner, repo, branch and path (this allows you to open a subfolder as a project)
- Adds a simple redirect from `/github.com/{owner}/{repo}/tree/{branch}/{...path}` to the homepage with the query parameters already set. This allows anyone to simply postpone sveltelab.dev before github.com and open the current path in sveltelab.
- The same is possible also for `/github/{owner}/{repo}/tree/{branch}/{...path}` if someone wants to share it and don't like the .com at the end.
- It uses streamed promises (available on vercel edge) so that the page load is still snappy and than there's a loading indicator in the home page

It needs a github key to work so probably we should also add that to the readme

As an example this is your library in sveltelab

https://sveltelab-git-github-integration-sveltelab.vercel.app/github.com/SarcevicAntonio/as-comps

### CODE REVIEW

File: src/routes/(repl)/+layout.server.ts [hunk 1]
```diff
@@ -13,17 +14,72 @@ async function get_repl_from_id(id: string, pocketbase: PoketBase) {
...
+	locals,
+	url: { searchParams },
+	cookies,
+}) => {
 	const { repl } = params;
+	const owner = searchParams.get('owner');
+	const repo = searchParams.get('repo');
+	const branch = searchParams.get('branch');
+	const path = searchParams.get('path');
+	let name = 'Hello world';
```
[the remark this answers is not in the document]
Comment (paoloricciuti) [reply src/routes/(repl)/+layout.server.ts:58]: Mmm good idea

File: src/routes/(repl)/+layout.svelte [hunk 2]
```diff
@@ -23,32 +23,46 @@
...
+				window.localStorage.removeItem(PUBLIC_SAVE_IN_LOCAL_STORAGE_NAME);
+			} else {
+				// check if there's the hash
+				const hash = window.location.hash.substring(1);
+				if (hash) {
+					const url_search_params = new URLSearchParams(hash);
+					const code = url_search_params.get('code');
+					if (code) {
+						const project = decompressFromEncodedURIComponent(code);
+						try {
```
[the remark this answers is not in the document]
Comment (paoloricciuti) [reply src/routes/(repl)/+layout.svelte:60]: definitely a good idea :D

**Comment by SarcevicAntonio** [comment]
im also not sure about the routing... I'd like to have a ?provider=github so we are open to other providers

**Comment by paoloricciuti** [comment]
> im also not sure about the routing... I'd like to have a ?provider=github so we are open to other providers

It's fair, i'll add that to the redirect but i think being able to prepend sveltelab.dev on github can really be beneficial (it's the same thing stackblitz does)

**Pull request merged**